synapse/storage/databases/main/deviceinbox.py

```diff
@@ -101,6 +119,8 @@ def __init__(self, database: DatabasePool, db_conn, hs: "HomeServer"):
 
     def process_replication_rows(self, stream_name, instance_name, token, rows):
         if stream_name == ToDeviceStream.NAME:
+            # If replication is happening than postgres must be being used.
+            assert isinstance(self._device_inbox_id_gen, MultiWriterIdGenerator)
```

Comment:
This is rather unfortunate since this code can only be called if replication is in use, but not sure if there's a "better" way to do it.

Reply:
I can't think of a better way either!